pkg/nar/types.go

```diff
@@ -14,6 +21,8 @@ const (
 	TypeSymlink = NodeType("symlink")
 )
 
+var NodeNameRegexp = regexp.MustCompile(fmt.Sprintf("^%v$", nixpath.NameRe))
```

Comment:
It seems like an implementation detail that should stay private to the module.

Reply:
Hmmh, it's also something that `pkg/nar/ls` could be using to verify the entry keys with. I'd prefer to keep it public.